lib/cache/role_test.go

```diff
@@ -21,11 +21,24 @@ import (
 	"time"
 
 	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/assert"
 
 	"github.com/gravitational/teleport/api/client/proto"
 	"github.com/gravitational/teleport/api/types"
 )
 
+func TestRoleNotFound(t *testing.T) {
+	t.Parallel()
+
+	p := newTestPack(t, ForNode)
+	t.Cleanup(p.Close)
+
+	r, err := p.cache.GetRole(t.Context(), "test-role")
+	assert.Nil(t, r)
```

Comment:
Unless otherwise specified by the function contract, the return value should not be inspected or assumed to be anything if there's an error. Is this a necessary check?